### ISSUE TITLE
Small improvements to custom objectives example [documentation]

### ISSUE THREAD

**Issue by e-pet** [pyro-ppl/pyro#3121]
The [Custom Objectives Example page](https://pyro.ai/examples/custom_objectives.html) currently states that svi.step() is equivalent to the following low-level code pattern "assuming we know all the parameters in advance":

```
# define optimizer and loss function
optimizer = torch.optim.Adam(my_parameters, {"lr": 0.001, "betas": (0.90, 0.999)})
loss_fn = pyro.infer.Trace_ELBO().differentiable_loss
# compute loss
loss = loss_fn(model, guide, model_and_guide_args)
loss.backward()
# take a step and zero the parameter gradients
optimizer.step()
optimizer.zero_grad()
```

The "assuming we know all the parameters in advance" part is not particularly helpful if one wants to actually do this, and TBH it is also not immediately obvious to me how I would get those automatically. I see that svi.step() has this:

```
with poutine.trace(param_only=True) as param_capture:
    loss = self.loss_and_grads(self.model, self.guide, *args, **kwargs)

params = set(site["value"].unconstrained()
             for site in param_capture.trace.nodes.values())
```

... but that seems to get the current parameter values, and I'm unsure whether that is what torch.optim.Adam requires?

In any case, I think it would be very helpful to really have an exact equivalent of doing

```
for i in range(0, n_iterations):
    svi.step()
```

in the example.

Finally, as a side-note, maybe this page should be renamed to "Custom objectives and training loops" or something like that, because it actually isn't only about modifying the objective? E.g., clipping gradients and mixing optimizers is not really about the objective. I was looking for information on customizing the training loop and only found this page by accident.

**Comment by fritzo** [comment]
Hi @e-pet, both of your suggestions sound very reasonable and we would be happy to accept a pull request :+1:

**Comment by martinjankowiak** [comment]
closed in #3122
